Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -332,116 +277,62 @@ open class NavigationMapView: UIView {
         addSubview(imageView)
     }
     
-    open override func layoutSubviews() {
-        super.layoutSubviews()
-        
-        // If the map is in tracking mode, make sure we update the camera and anchor after the layout pass.
-        if tracksUserCourse {
-            updateCourseTracking(location: userLocationForCourseTracking, camera:mapView.cameraView.camera, animated: false)
-            
-            // TODO: Find appropriate place where anchor can be updated.
-            mapView.cameraView.anchor = userAnchorPoint
-        }
-    }
-    
     /**
      Updates the map view’s preferred frames per second to the appropriate value for the current route progress.
      
-     This method accounts for the proximity to a maneuver and the current power source. It has no effect if `tracksUserCourse` is set to `true`.
+     This method accounts for the proximity to a maneuver and the current power source.
+     It has no effect if `NavigationCameraState` is in `.following` mode.
      */
-    open func updatePreferredFrameRate(for routeProgress: RouteProgress) {
-        guard tracksUserCourse else { return }
+    public func updatePreferredFrameRate(for routeProgress: RouteProgress) {
```

Comment:
I had to slightly rework frame rate changing functionality. After adding new camera animations we have to always update `UserCourseView` frame by frame. At the same time I'm not sure why `pluggedInFramesPerSecond` is always set to `PreferredFPS.lowPower`, I'd expect it to be `PreferredFPS.maximum`.

Reply:
Yes, but in current implementation we no longer require `shouldPositionCourseViewFrameByFrame`, as `UserCourseView` is always updated on each frame change. I think in #2643 we incorrectly use `pluggedInFramesPerSecond`, as it's always set to `MGLMapViewPreferredFramesPerSecond.lowPower`, which is not 30 FPS. I think in current PR right thing would be to use `PreferredFPS.maximum` for `pluggedInFramesPerSecond`.

/cc @Udumft

Reply:
TBH, I tried to recall why exactly `defaultFramesPerSecond` and `pluggedInFramesPerSecond` were set as they were, but couldn't...
For implementation in this PR I have a question: we are opting to select `.normal` frame rate unless we encounter a straight area of a route or are going away from maneuver nodes, at which case we are a) ramping up frame rate if device is plugged in or b) slowing down to `.lowPower` otherwise. Why same event causes reverse FPS reaction? Shouldn't we always increase or decrease frame rate at such event?

Reply:
I'm not too familiar with specifics/reasoning of underlying logic, but `a` seems to be reasonable. In case of `b` (based on current implementation) we're not necessarily changing frame rate to `PreferredFPS.lowPower`, it's up to user to decide how it can be modified in `minimumFramesPerSecond`.

On the other hand I'm not sure why we change `PreferredFPS` based on such maneuver directions (e.g. one of the cases checks only **slight** [.straightAhead, .slightLeft, .slightRight] directions, I'd expect to increase frame rate when we're approaching a **sharp** maneuver).

In any case this doesn't directly change public APIs so we can tweak it as a tail work if such need arises.

/cc @1ec5

Reply:
> On the other hand I'm not sure why we change `PreferredFPS` based on such maneuver directions (e.g. one of the cases checks only **slight** [.straightAhead, .slightLeft, .slightRight] directions, I'd expect to increase frame rate when we're approaching a **sharp** maneuver).

The original intention was to animate as smoothly as possible when making a turn, because any throttling would be perceived as choppy animation. But straight and slight turns don’t cause the camera to swing around, so there’s no need to disable throttling in those cases.